### ISSUE TITLE
[Epic] Elemental teal acceptance tests on upgrades

### ISSUE THREAD

**Issue by davidcassany** [rancher/elemental#651]
We need to define the acceptance test matrix for an Elemental Teal upgrade.

The outcome of this issue is a list of all the tests that any new version of the elemental-teal image must pass before being promoted to `Stable` area in OBS. We should also consider how actually run the tests as part of this issue (ensure we have all the pieces to, at least, manually run them).

My suggestion are the following tests, the image to test should living in `Staging` area:

1. Clean install (no upgrades, using the ISO) with the **latest** elemental-operator on the management cluster.
2. Clean install (no upgrades, using the ISO) with the **oldest** stable elemental-operator on the management cluster.
3. Upgrade using a `ManagedOSVersionChannel` with the **latest** elemental-operator on the management cluster from the **latest** stable elemental-teal image in the node.
4. Upgrade using a `ManagedOSVersionChannel` with the **oldest** elemental-operator on the management cluster from the **oldest** stable elemental-teal image in the node.

Any further thoughts or ideas?

Steps to do for the new upgrade tests:
- [x] disable current upgrade tests from CI (not useful anymore) #668  and https://github.com/rancher/elemental/pull/679
- [x] remove upgrade with elemental CLI as it's not something supported (can be added in elemental-toolkit if needed/not already done) #669
- [x] re-add a daily CI upgrade tests from current (or "old") Stable version to HEAD version (on a small 3 nodes cluster) #670
- [x] re-add Verification upgrade tests from current (or "old") Stable version to Dev/Staging version (tests done before a new release, not scheduled) #671
- [x] UI: test upgrade in a dedicated test #659

**Comment by davidcassany** [comment]
@ldevulder @juadk I think that after #649 we have all the pieces to start exercising upgrades tests in meaningful way. At least all this should be possible to do manually with the current builds available.

According to the list I did above I believe we should differentiate the upgrade tests from a clean deployment test as they have a different starting point environment.

So I would expect for each elemental teal release to run:

1. e2e tests as we have now from staging, without any node upgrade, just to test clean installations with the upcoming elemental-teal and elemental-operator pair.
2. The same as above but using the **oldest** elemental-operator instead. (We will make it available in current `Stable` project, I have been thinking on locking the current one there to keep the first release around for testing)
3. Deploy a cluster from stable environment (using **oldest** elemental-operator), add the test channel as described in https://github.com/rancher/elemental/issues/649#issuecomment-1422873628 and run an OS upgrade to `staging` image. I guess we should verify the upgrade happens without down time as fleet should orchestrate the upgrade to make it possible.
4. The same as above but using the upcoming or latest elemental-operator.

From the current e2e tests I guess we should differentiate upgrade tests from clean deployment tests and decouple elemental-teal version from elemental-operator, it should be possible to use `stable` elemental-operator with `staging` elemental-teal.

Does all that make sense you?

**Comment by ldevulder** [comment]
Looks good to me. The current code for E2E tests easily allows upgrade tests to be added/removed and it also allows the mix of elemental-operator and elemental-teal versions, so what you ask should be doable without major issue I think.

Maybe the workflow for an upgrade test scenario could be to provision a cluster (or more) with 3 nodes and do the different upgrades scenarii you asked. And for the daily CI we can use a new test where we install a stable version of all and upgrade to the dev ones (to try to catch any issues ASAP).

**Comment by ldevulder** [comment]
So what we could do for the daily CI:
- running deployments tests in CLI and UI from latest HEAD/Dev without any upgrades (with all nodes)
- running new upgrade tests in CLI and UI from `Stable` and upgrade to latest HEAD/Dev (with few nodes, 1 or 3)

So what we could do for the Validation CI (following your order):
- running the deployment test on full cluster (with all nodes)
- running the upgrade tests on smaller cluster (with few nodes, 1 or 3)

**Comment by juadk** [comment]
We talked about it with @ldevulder during a slack huddle so all good to me as well.

**Comment by juadk** [comment]
I think we can close the epic, last topic is covered by https://github.com/rancher/elemental/issues/696 and it can not be done for now. However, we also talked about pushing an application in 3 nodes cluster and check we can upgrade without any downtime, I also think we can open a separate issue for that case. @davidcassany @ldevulder WDYT?

**Comment by ldevulder** [comment]
@juadk yes a separate issue for this use case is a good idea. It's clearly something to have in our test IMHO.
